hack/local-up-volcano.sh

```diff
@@ -0,0 +1,137 @@
+CLUSTER_CONTEXT="--name ${CLUSTER_NAME}"
+KIND_OPT=${KIND_OPT:-}
+INSTALL_MODE=${INSTALL_MODE:-"kind"}
+IMAGE_PREFIX=volcanosh/vc
+TAG=${TAG:-`git rev-parse --verify HEAD`}
+RELEASE_DIR=_output/release
+RELEASE_FOLDER=${VK_ROOT}/${RELEASE_DIR}
+YAML_FILENAME=volcano-${TAG}.yaml
+
+
+# check if kind installed
+function check-prerequisites {
+  echo "checking prerequisites"
+  which kubectl >/dev/null 2>&1
+  if [[ $? -ne 0 ]]; then
+    echo "kubectl not installed, exiting."
+    exit 1
+  else
+    echo -n "found kubectl, " && kubectl version --short --client
+  fi
+}
+
+# check if kind installed
+function check-kind {
+  echo "checking kind"
+  which kind >/dev/null 2>&1
+  if [[ $? -ne 0 ]]; then
+    echo "kind not installed, exiting."
+    exit 1
+  else
+    echo -n "found kind, version: " && kind version
+  fi
+}
+
+# prepare deploy yaml and docker images
+function prepare {
+  echo "Preparing..."
+  echo "Generating valcano deploy yaml"
+  make generate-yaml
+
+  echo "Building docker images"
+  make images
+}
+
+# spin up cluster with kind command
+function kind-up-cluster {
+  check-kind
+  echo "Running kind: [kind create cluster ${CLUSTER_CONTEXT} ${KIND_OPT}]"
+  kind create cluster ${CLUSTER_CONTEXT} ${KIND_OPT}
+
+  echo "Loading docker images into kind cluster"
+  kind load docker-image ${IMAGE_PREFIX}-controllers:${TAG}  ${CLUSTER_CONTEXT}
+  echo kind load docker-image ${IMAGE_PREFIX}-controllers:${TAG}  ${CLUSTER_CONTEXT}
+  kind load docker-image ${IMAGE_PREFIX}-scheduler:${TAG}  ${CLUSTER_CONTEXT}
+  kind load docker-image ${IMAGE_PREFIX}-admission:${TAG}  ${CLUSTER_CONTEXT}
+}
+
+function install-volcano {
+  # TODO: add a graceful way waiting for all crd ready
+  kubectl apply -f ${RELEASE_FOLDER}/${YAML_FILENAME} --kubeconfig ${KUBECONFIG}
+  sleep 5
+  kubectl apply -f ${VK_ROOT}/installer/helm/chart/volcano/templates/default-queue.yaml --kubeconfig ${KUBECONFIG}
+}
+
+function uninstall-volcano {
+  kubectl delete -f ${VK_ROOT}/installer/helm/chart/volcano/templates/default-queue.yaml --kubeconfig ${KUBECONFIG}
+  kubectl delete -f ${RELEASE_FOLDER}/${YAML_FILENAME} --kubeconfig ${KUBECONFIG}
+}
+
+# clean up
+function cleanup {
+  uninstall-volcano
+
+  if [ "${INSTALL_MODE}" == "kind" ]; then
+    echo "Running kind: [kind delete cluster ${CLUSTER_CONTEXT}]"
+    kind delete cluster ${CLUSTER_CONTEXT}
+  fi
+}
+
+echo $* | grep -E -q "\-\-help|\-h"
+if [[ $? -eq 0 ]]; then
+  echo "Customize the kind-cluster name:
+
+    export CLUSTER_NAME=<custom cluster name>  # default: volcano
+
+Customize kind options other than --name:
+
+    export KIND_OPT=<kind options>
+
+Using existing kubernetes cluster rather than starting a kind custer:
+
+    export INSTALL_MODE=existing
+
+Cleanup all installation:
+
+    ./hack/local-up-volcano.sh -q
+"
+  exit 0
+fi
+
+
+echo $* | grep -E -q "\-\-quit|\-q"
+if [[ $? -eq 0 ]]; then
+  cleanup
+  exit 0
+fi
+
+check-prerequisites
+
+prepare
+
+if [ "${INSTALL_MODE}" == "kind" ]; then
```

Comment:
For the clusters other than kind cluster, how can user deploy the images into the cluster?

Reply:
good question: we can not know their registry. If they use one node development env, the image is on the host. But if it is a multi-node cluster, we can not do that for users.

Reply:
Need to mention this in the document or just support kind cluster, since for the other cases, they can deploy volcano via the installation guide.

Reply:
SG